base/shared/src/main/scala/scalaz/tc/band.scala

```diff
@@ -0,0 +1,28 @@
+package scalaz.tc
+
+import scala.annotation.tailrec
+
+/**
+ * Band is a semigroup with idempotency.
+ *
+ */
+trait BandClass[A] extends SemigroupClass[A] {
+
+  /**
+   * For `n = 0`, `value`
+   * For `n = 1`, `append(value, value)`
+   * For `n = 2`, `append(append(value, value), value)`
+   *
+   * The default definition uses peasant multiplication, exploiting associativity to only
+   * require `O(log n)` uses of [[mappend]]
+   */
+  def multiply1(value: A, n: Int): A = {
```

Comment:
Doesn't this always return `value`, due to idempotency?

Reply:
Yes. But do we want to exclusively compute the idempotent behavior or do we just want say "hey, this is an idempotent semigroup" and have it be a marker trait for the extension?